ingress/controllers/nginx/README.md

```diff
@@ -0,0 +1,184 @@
+# Nginx Ingress Controller
+
+This is a nginx Ingress controller that uses [ConfigMap](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/configmap.md) to store the nginx configuration. See [Ingress controller documentation](../README.md) for details on how it works.
+
+
+## What it provides?
+
+- Ingress controller
+- nginx 1.9.x with
+- SSL support
+- custom ssl_dhparam (optional). Just mount a secret with a file named `dhparam.pem`.
+- support for TCP services (flag `--tcp-services-configmap`)
+- custom nginx configuration using [ConfigMap](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/configmap.md)
+- custom error pages. Using the flag `--custom-error-service` is possible to use a custom compatible [404-server](https://github.com/kubernetes/contrib/tree/master/404-server) image
+
+
+## Requirements
+- default backend [404-server](https://github.com/kubernetes/contrib/tree/master/404-server)
+
+
+
+## Create a loadbalancer
+
+Loadbalancers are created via a ReplicationController or Daemonset
+
+First we need to deploy some application to publish. To keep this simple we will use the [echoheaders app](https://github.com/kubernetes/contrib/blob/master/ingress/echoheaders/echo-app.yaml) that just returns information about the http request as output
+```
+kubectl run echoheaders --image=gcr.io/google_containers/echoserver:1.1 --replicas=1 --port=8080
+```
+
+Now we expose the same application in two different services (so we can create different Ingress rules)
+```
+kubectl expose rc echoheaders --port=80 --target-port=8080 --name=echoheaders-x
+kubectl expose rc echoheaders --port=80 --target-port=8080 --name=echoheaders-y
+```
+
+Next we create a couple of Ingress rules
+```
+kubectl create -f examples/ingress.yaml
+```
+
+we check that ingress rules are defined:
+```
+$ kubectl get ing
+NAME      RULE          BACKEND   ADDRESS
+echomap   -
+          foo.bar.com
+          /foo          echoheaders-x:80
+          bar.baz.com
+          /bar          echoheaders-y:80
+          /foo          echoheaders-x:80
+```
+
+Before the deploy of the Ingress controller we need a default backend [404-server](https://github.com/kubernetes/contrib/tree/master/404-server)
+```
+kubectl create -f examples/default-backend.yaml
```

Comment:
Lets keep this section *just* about creating the loadblancer. like so:

## Deploy the Ingress controller

$ kubectl create -f all-in-one.yaml

## HTTP 

kubctl create echoserver...

## TLS

## TCP

## Troubleshooting

### Custom Errors 

### Debugging

## Limitations